zookeeper-jute/src/main/java/org/apache/jute/BinaryInputArchive.java

```diff
@@ -85,6 +85,14 @@ public String readString(String tag) throws IOException {
     }
     
     static public final int maxBuffer = Integer.getInteger("jute.maxbuffer", 0xfffff);
+    static public int readExtraSize = Integer.getInteger("zookeeper.jute.maxbuffer.extrasize",
+            maxBuffer);
+    static {
+        // Earlier hard coded value is 1024, So the value should not be less than that
+        if (readExtraSize < 1024) {
```

Comment:
Assign an initial value to a temp variable here, do this small computation, then assign the value to the variable.
It is legal to assign a value in a static block to a final static field, the only constraint is that you do the assignement only once

Reply:
ok, changed it